kahuna/public/js/search/results.js

```diff
@@ -281,7 +279,9 @@ results.controller('SearchResultsCtrl', [
 
             // Default explicit until/since to $stateParams
             if (angular.isUndefined(until)) {
-                until = $stateParams.until || lastSearchFirstResultTime;
+                // if we have a `lastSearchTime`, use that instead of `$stateParams` so that if the
+                // stateParams.until was in the future, we're not loading a mutating set of results.
+                until = lastSearchFirstResultTime || $stateParams.until;
```

Comment:
So this is based on the assumption that `lastSearchFirstResultTime` is always earlier than `$stateParams.until`, because `$stateParams.until` would have been used the first time to fetch the results which resulted in `lastSearchFirstResultTime` being set?

Makes sense, but given how fiddly that rationale is, and brittle in the face of change, might be worth expliciting it here in the comments so the next person (who may well be us :-) ) doesn't have to think for too long.